packages/mattermost-redux/src/selectors/entities/channels.ts

```diff
@@ -851,6 +868,32 @@ export const getUnreadChannelIds: (state: GlobalState, lastUnreadChannel?: Chann
     },
 );
 
+export const getAllTeamsUnreadChannelIds: (state: GlobalState, lastUnreadChannel?: Channel | null) => string[] = createIdsSelector(
+    'getAllTeamsUnreadChannelIds',
+    isCollapsedThreadsEnabled,
+    getMyChannelMemberships,
+    getChannelMessageCounts,
+    getChannelIdsForAllTeams,
+    (state: GlobalState, lastUnreadChannel: Channel | undefined | null = null): Channel | undefined | null => lastUnreadChannel,
+    (
+        collapsedThreads,
+        members: RelationOneToOne<Channel, ChannelMembership>,
+        messageCounts: RelationOneToOne<Channel, ChannelMessageCount>,
+        allTeamsChannelIds: string[],
+        lastUnreadChannel?: Channel | null,
+    ): string[] => {
+        const unreadIds = allTeamsChannelIds.filter((id) => {
+            return calculateUnreadCount(messageCounts[id], members[id], collapsedThreads).showUnread;
+        });
+
+        if (lastUnreadChannel && members[lastUnreadChannel.id] && !unreadIds.includes(lastUnreadChannel.id)) {
```

Comment:
@hmhealey We are using LastUnreadChannel here, which we use in ``getUnsortedAllTeamsUnreadChannels`` and that is getting used in ``getSortedAllTeamsUnreadChannels``. What do you suggest should I remove LastUnread from here as well.

Reply:
Yeah, we should remove it since this is only used for the channel switcher. `lastUnreadChannel` is only needed for selectors that are used in the sidebar.